### DISCUSSION TITLE
Consumer getting stuck when messages sending at 0.2 seconds

### DISCUSSION THREAD

**Discussion by Pramod34** [rabbitmq/rabbitmq-dotnet-client#1754] [Q&A]
We have applications where host/client/producer publish messages to topic exchange, based on routing key messages are sent to respective queue and Engines/servers/consumers will read messages from queue, process and sends reply back to host/client via different queue. Host/client will consume the reply message.

When running application to send message at every 1 second, everything is working fine. But when we are running an application to send message at every 0.5 or 0.2 second, it is working(host/client able to receive reply messages) for few messages, but suddenly reply receiving consumer is getting strucked and unable to read the messages, messages are struck in queue until request receiver(Engine/server) reads all the messages. Once engines completes all messages, then reply receiver (host consumer) reads the messages.

We are using .NET RabbitMQ.Client library with EventingBasicConsumer method for consuming messages from RabbitMQ. I have check RabbitMQ logs, RabbitMQ connection and no memory issue. Everything seems to be okay, when reply receiver unable to consume messages from RabbitMQ queue. EventingBasicConsumer is getting triggered after struck, even though messages are in queue.

I would be really thankful and appreciate any solution for my problem.

**Comment by lukebakken** [comment]
Hello @Pramod34,

Your GitHub history shows that you have received quite a bit of free support from Team RabbitMQ. Notable `RabbitMQ.Client` questions include the following:
* https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions/1584
* https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions/1461
* https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions/1650#discussioncomment-10246928

In some of these cases, _I_ spent time putting together code to demonstrate a fix for your issue.

Let me remind you that to be eligible for free community support, you must meet [our guidelines](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#who-is-eligible-for-community-support):
* Use the latest version of RabbitMQ (4.x)
* Use a recent version of this library (6.x or 7.x)
* Provide a _DETAILED_ issue report.

I'm willing to continue assisting you with the issue you report here, but you _MUST_ provide a .NET project I can clone, compile, and run, to see the same behavior you report. I am certain you are not using this library correctly, but I do not have the time to try and guess how to reproduce what you report.

In short, these are the steps I should have to do to reproduce the issue you report here (discussion #1754):

```
git clone https://github.com/Pramod34/rabbitmq-dotnet-client-1754.git
cd rabbitmq-dotnet-client-1754
dotnet build
dotnet run
```

Thanks

**Comment by michaelklishin** [comment]
@Pramod34 we will not debug your applications for you, and we won't guess what your applications do exactly, what RabbitMQ logs, what the metrics look like when nodes log certain things, and so on.

The most common scenario is a consumer that [uses manual acknowledgements](https://www.rabbitmq.com/docs/consumers#acknowledgement-modes) and a [limited prefetch](https://www.rabbitmq.com/docs/confirms#channel-qos-prefetch), then fails to acknowledge deliveries (send a `basic.ack` frame) for any reason, or acknowledges the same delivery twice, or on the wrong channel, resulting in a [channel exception](https://www.rabbitmq.com/docs/channels#error-handling) which it then handles but RabbitMQ will certainly [log](https://www.rabbitmq.com/docs/logging).

It's very easy to verify this hypothesis [with Wireshark](https://www.rabbitmq.com/amqp-wireshark).

In theory a long running consumer could run into a [consumer delivery timeout](https://www.rabbitmq.com/docs/consumers#acknowledgement-timeout) but that timeout is 30 minutes by default, so given the brief problem definition I suspect that the test run does not take long enough.
